src/Rules/PhpDoc/RequireExtendsCheck.php

```diff
@@ -41,39 +43,48 @@ public function checkExtendsTags(Node $node, array $extendsTags): array
 
 		foreach ($extendsTags as $extendsTag) {
 			$type = $extendsTag->getType();
-			if (!$type instanceof ObjectType) {
+			if (count($type->getObjectClassNames()) === 0) {
 				$errors[] = RuleErrorBuilder::message(sprintf('PHPDoc tag @phpstan-require-extends contains non-object type %s.', $type->describe(VerbosityLevel::typeOnly())))
 					->identifier('requireExtends.nonObject')
 					->build();
 				continue;
 			}
 
-			$class = $type->getClassName();
-			$referencedClassReflection = $type->getClassReflection();
+			$classNames = $type->getObjectClassNames();
+			sort($classNames);
```

Comment:
I am sorting this array because tests seem to fail randomly.

 * https://github.com/phpstan/phpstan-src/actions/runs/13726816592/job/38395109333?pr=3858

<img width="312" alt="" src="https://github.com/user-attachments/assets/b297c86e-7c52-4fb8-99a1-2c98f7519903" />
<img width="1119" alt="" src="https://github.com/user-attachments/assets/72d75293-2914-45ba-94c1-07eddbbf6e9b" />